WooCommerce/Classes/ViewRelated/Products/ProductsListViewModel.swift

```diff
@@ -151,3 +160,27 @@ class ProductListViewModel {
         stores.dispatch(batchAction)
     }
 }
+
+// MARK: - Blaze banner visibility
+extension ProductListViewModel {
+    /// Checks for Blaze eligibility and user defaults to show the banner if necessary.
+    ///
+    @MainActor
+    func updateBlazeBannerVisibility() async {
+        let isSiteEligible = await blazeEligibilityChecker.isSiteEligible()
+        guard isSiteEligible else {
+            return
+        }
+        guard userDefaults[.hasDismissedBlazeBanner] == nil else {
+            return
```

Comment:
❓ when reaching the else cases after the first check (since `updateBlazeBannerVisibility` can be called more than once) like when the site becomes ineligible or the user defaults has been reset for some reason, do we want to set `shouldShowBlazeBanner` to `false` as well?

it might also be safer to have a private function that returns the eligibility async to be set to `shouldShowBlazeBanner` since a bool value is returned for each path.

Reply:
Also updated in 26bbf0e8368689d07c4d0b8de9e3231206c5e9f3.